### DISCUSSION TITLE
Adding Slash Lighting Functionality

### DISCUSSION THREAD

**Discussion by lgarlati** [seerge/g-helper#2687] [Ideas]
Followup from my previous post (https://github.com/seerge/g-helper/discussions/2680) and mostly a checklist for myself. But since most of these aren't implemented yet, "ideas" category makes the most sense to me. Also, I may not implement all of these. If anyone wants one of these, please reply and I'll make sure to implement it. 

Planning to add:
- Battery Charge State Pattern
   - ~Basic pattern setting and updating~
   - Battery Charging animation until charged to full
   - Show any other pattern once charged to full
- Auto dim
   - Have slash brightness follow screen brightness
- Dim on battery
   - Once on battery, dim slash instead of just turning it off

Not slash lighting, but something I think would be great is flicker free dimming with windows dimming. Something like:
- once windows default dimming has reached minimum, begin dimming via flicker free
- dim via flicker free first, then via windows default dimming
- whenever windows dim button is pressed, dim via flicker free an equivalent amount

**Comment by seerge** [comment]
@lgarlati hello, thanks for your initiative! I have merged your PR.

As for other things you have listed. 

It's cool - but I'm just afraid it's too niche / specific. You can make it for yourself in your own fork of course. But on this stage I don't plan to add that to the app, cause I'm trying to keep it as simple to use as possible (and it already has too many different features)

**Reply by lgarlati** [reply]
Makes sense, and will do.

If you need help with anything in the future, give me an @ and I may be able to give assistance. 

For future reference, if you need help testing, I have a GU605MZ (2024 g16 4080)

**Reply by seerge** [reply]
@lgarlati sure :) G16 is amazing device. Do all core features of G-Helper work as they are intended ? :)

**Reply by seerge** [reply]
@lgarlati oh, and since you have a similar device - can you please check if Gamut / Dimming behavior explained in this issue https://github.com/seerge/g-helper/issues/2543#issuecomment-2097187320 is indeed the case ? (you can comment there directly)

**Reply by lgarlati** [reply]
> Do all core features of G-Helper work as they are intended ? :)

As far as I can tell, yeah. Slash lighting recently being fixed in the last release.

> can you please check if Gamut / Dimming behavior explained in this issue https://github.com/seerge/g-helper/issues/2543#issuecomment-2097187320

Replied, seems to be fine for me

**Reply by HoneyBear4Lyfe** [reply]
You two are great. I love that I can finally turn off the slash lighting when the lid is closed without completely powering down my G16. I get that the lights are cool but I just want them to be completely disabled since I primarily use my laptop in the dark. This brings me to my further request-
Would it be possible to add the option to disable slash lighting when the laptop _falls asleep_? I often use it in the dark, going to bed watching movies or playing music. I wake up in the middle of the night to the whole room strobing from the LEDs on the back of my computer's screen. Would it be possible to change the "disable on lid close" to be "disable on sleep"? I think this makes sense, since laptops just go into sleep mode anyways when the lid closes.
Alternately, is it possible to 100% disable slash lighting? If my suggestion is not a setting you're willing or able to implement, do you know if I can just entirely delete whatever it is on the G16 that turns the light on in the first place?
Thanks for all your time on this project! I love that I was able to delete the awful bloatware that this fancy computer came with!

**Reply by seerge** [reply]
> Would it be possible to add the option to disable slash lighting when the laptop falls asleep? I often use it in the dark, going to bed watching movies or playing music. I wake up in the middle of the night to the whole room strobing from the LEDs on the back of my computer's screen. Would it be possible to change the "disable on lid close" to be "disable on sleep"? I think this makes sense, since laptops just go into sleep mode anyways when the lid closes.

That's not possible because when laptop is about to get asleep windows suspends all processes, including G-Helper

> Alternately, is it possible to 100% disable slash lighting?

This is already the case, just set brightness to 0% and it will be permanently OFF

**Reply by HoneyBear4Lyfe** [reply]
Ahh, that makes a lot of sense.
Sweet! thanks for your immediate response!

**Reply by lgarlati** [reply]
@HoneyBear4Lyfe I know this isn't what you were asking for, but having the slash lighting correspond to panel brightness may help. Ie in the dark, the slash dims as your dim your panel. So even if it stays on, it's not full blast. This is currently possible via the shortcut, but personally, I prefer automation over hotkey combos. 
I've been wanting to implement this for personal use, but haven't found a compelling enough reason for the effort. If you think this may be helpful, I'll take a look into if it is possible.

@seerge I know we discussed this already, and concluded this was somewhat unnecessary given the hotkey. But maybe it's worth reconsidering? I know the main issue with extra features is clutter and bloat, but I don't think adding an "auto" brightness option to the brightness dropdown is too egregious? As for bloat, no clue. Do you think there is a way to get an interrupt when windows brightness is adjusted? Polling every minute would be both inefficient and annoying to the user.
